.github/workflows/ci.yaml

```diff
@@ -21,7 +21,7 @@ jobs:
       matrix:
         os: [ubuntu-latest, macOS-latest, windows-latest]
         rustalias: [stable, nightly, msrv]
-        feature_flag: ["--all-features", "--no-default-features", ""]
+        feature_flag: ["--no-default-features --features sync_all", "--no-default-features --features tokio_all", "--no-default-features --features sync", "--no-default-features --features tokio", ""]
```

Comment:
Could you please make it possible to build with both `sync` and `tokio`? To do this, you can invoke the traits with an explicit self parameter, e.g. `Read::read_exact(self, buf)` and `AsyncRead::read_exact(self, buf)` instead of `self.read_exact(buf)`.

Reply:
FYI, I found a crate that does something similar to this, but with proc macros: https://crates.io/crates/async-generic